File: algorithms/quicksort.js
function quickSort(arr){
    if(arr.length < 2) {
        return arr
    }
    else {
        pivot = arr[0]
        greater = [...arr].splice(1).filter(item => (item > pivot))
        smaller = [...arr].splice(1).filter(item => (item <= pivot))
        sortedSmaller = quickSort(smaller)
        sortedGreater = quickSort(greater)
        return [...sortedSmaller, pivot, ...sortedGreater]
    }
}

arr = [12,2,55,9,1,4,5,30]
console.log(quickSort(arr))

File: algorithms/quicksort.py
def quicksort(array):
    if(len(array) < 2):
        return array
    else:
        pivot = array[0]
        less = [i for i in array[1:] if i <= pivot]
        bigger = [i for i in array[1:] if i > pivot]
        return quicksort(less) + [pivot] + quicksort(bigger)

arr = [12,2,55,9,1,4,5,30]
print(quicksort(arr))